phoenix-core/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java

```diff
@@ -178,8 +179,8 @@ public MutationPlan compile(CreateTableStatement create) throws SQLException {
                     }
                 }
                 if (isPKMissed) {
-                    throw new SQLExceptionInfo.Builder(SQLExceptionCode.PRIMARY_KEY_MISSING)
-                            .build().buildException();
+                    throw new TableNotFoundException(tableRef.getTable().getSchemaName().toString(),
```

Comment:
Are there cases where the original exception would be the correct one that we need to distinguish?